File: part_2-regression/random-forest-regression/random_forest_regression.r
#  Random Forest Regression 

#   Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

#   Splitting the data into Training Set and Testing Set
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Salary, SplitRatio = 2/3)
# training_set = subset(dataset, split == TRUE)
# test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)

#   Fitting the Random Forest Regression to the dataset
#install.packages('randomForest')
library(randomForest)
set.seed(1234)
regressor = randomForest(x = dataset[1],
                         y = dataset$Salary,
                         ntree = 500)

#   Predicting a new result with Random Forest Regression
y_pred = predict(regressor, data.frame(Level = 6.5))


#   Visualizing the Random Forest Regression results (Higher resolution and Smoother Curve)
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)

ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = 'red') +
    geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))), colour = 'blue') +
    ggtitle('Truth or Bluff (Regression Model') +
    xlab('Level') +
    ylab('Salary')

File: part_2-regression/random-forest-regression/random_forest_regression.py
# Random Forest Regression

# Import the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd 

#   Importing Dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

#Splitting the dataset into Training set and Test set
"""from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = .2, random_state =0)"""

#   Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

#   Fitting the Regression to the dataset
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators = 300, random_state = 0)
regressor.fit(X, y)

#   Predicting a new result with Regression Model
y_pred = regressor.predict(6.5)


#   Visualizing the Regression Model results (for higher resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (Random Forest Regression -> 10 estimators)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()


